scripts/west_commands/build.py

```diff
@@ -106,6 +111,17 @@ def do_add_parser(self, parser_adder):
         parser.add_argument('-t', '--target',
                             help='''Override the build system target (e.g.
                             'clean', 'pristine', etc.)''')
+        parser.add_argument('-p', '--pristine', choices=['auto', 'always',
+                            'never'], action=AlwaysIfMissing, nargs='?',
+                            help='''Control whether the pristine target is run
+                            before building if a build system is present in the
+                            build dir. --pristine is the same as
+                            --pristine=always. If set to auto, the pristine
```

Comment:
No equal sign, right? ``--pristine always``

Reply:
both are correct, `--pristine always` and `--pristine=always`. I used the one with the equal sign to make it fool-proof.